File: packages/rune-animate/ios/RuneAnimate.swift
//
//  RuneAnimate.swift
//  RuneAnimate
//
//  Public interface for RuneAnimate module
//

import Foundation
import RuneKit

@objc(RuneAnimate)
public final class RuneAnimate: NSObject {
  private static var isInstalled = false

  @objc public static func initialize(with runtime: RuneRuntime) {
    if isInstalled {
      print("[RuneAnimate] Module already initialized")
      return
    }

    runtime.installModules([RuneAnimateBridge(runtime: runtime)])
    isInstalled = true
    print("[RuneAnimate] Module initialized")
  }
}

File: packages/rune-animate/android/RuneAnimate/src/main/java/dev/rune/animate/RuneAnimate.kt
package dev.rune.animate

import com.rune.kit.runtime.RuneRuntime

/**
 * Public interface for RuneAnimate module.
 */
object RuneAnimate {
    private var moduleInstance: RuneAnimateModule? = null

    @JvmStatic
    fun initialize(runtime: RuneRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("RuneAnimate", "Module already initialized")
            return
        }

        moduleInstance = RuneAnimateModule(runtime)
        runtime.installModules(listOf(moduleInstance!!))
        android.util.Log.d("RuneAnimate", "Module initialized")
    }

    @JvmStatic
    fun cleanup() {
        moduleInstance?.onDestroy()
        moduleInstance = null
    }
}
